esphome/core/application.cpp

```diff
@@ -108,6 +90,21 @@ void Application::loop() {
     this->components_[this->dump_config_at_]->call_dump_config();
     this->dump_config_at_++;
   }
+
+  const uint32_t elapsed = millis() - this->last_loop_;
+  uint32_t delay_time = 1;  // minimum delay due to tasks at prio 0
```

Comment:
Now that the lower-priority tasks have been eliminated by the other PRs, I would see no issue with  just setting this to 0.
```suggestion
  uint32_t delay_time = 0;
```